did_doc_builder/src/schema/did_doc.rs

```diff
@@ -0,0 +1,323 @@
+    extra: HashMap<String, Value>,
+}
+
+impl DidDocumentBuilder {
+    pub fn new(id: Did) -> Self {
+        Self {
+            id,
+            ..Default::default()
+        }
+    }
+
+    pub fn add_also_known_as(mut self, also_known_as: Uri) -> Self {
+        self.also_known_as.push(also_known_as);
+        self
+    }
+
+    pub fn add_controller(mut self, controller: Did) -> Self {
+        self.controller.push(controller);
+        self
+    }
+
+    pub fn add_verification_method(mut self, verification_method: VerificationMethod) -> Self {
+        self.verification_method.push(verification_method);
+        self
+    }
+
+    pub fn add_authentication_method(mut self, method: VerificationMethod) -> Self {
+        self.authentication
+            .push(VerificationMethodKind::Resolved(method));
+        self
+    }
+
+    pub fn add_authentication_reference(mut self, reference: DidUrl) -> Self {
+        self.authentication
+            .push(VerificationMethodKind::Resolvable(reference));
+        self
+    }
+
+    pub fn add_assertion_method(mut self, method: VerificationMethod) -> Self {
+        self.assertion_method
+            .push(VerificationMethodKind::Resolved(method));
+        self
+    }
+
+    pub fn add_assertion_method_reference(mut self, reference: DidUrl) -> Self {
+        self.assertion_method
+            .push(VerificationMethodKind::Resolvable(reference));
+        self
+    }
+
+    pub fn add_key_agreement(mut self, key_agreement: VerificationMethod) -> Self {
+        self.key_agreement
+            .push(VerificationMethodKind::Resolved(key_agreement));
+        self
+    }
+
+    pub fn add_key_agreement_refrence(mut self, reference: DidUrl) -> Self {
+        self.key_agreement
+            .push(VerificationMethodKind::Resolvable(reference));
+        self
+    }
+
+    pub fn add_capability_invocation(mut self, capability_invocation: VerificationMethod) -> Self {
+        self.capability_invocation
+            .push(VerificationMethodKind::Resolved(capability_invocation));
+        self
+    }
+
+    pub fn add_capability_invocation_refrence(mut self, reference: DidUrl) -> Self {
+        self.capability_invocation
+            .push(VerificationMethodKind::Resolvable(reference));
+        self
+    }
+
+    pub fn add_capability_delegation(mut self, capability_delegation: VerificationMethod) -> Self {
+        self.capability_delegation
+            .push(VerificationMethodKind::Resolved(capability_delegation));
+        self
+    }
+
+    pub fn add_capability_delegation_refrence(mut self, reference: DidUrl) -> Self {
+        self.capability_delegation
+            .push(VerificationMethodKind::Resolvable(reference));
+        self
+    }
+
+    pub fn add_service(mut self, service: Service) -> Self {
+        self.service.push(service);
+        self
+    }
+
+    pub fn add_extra_field(mut self, key: String, value: Value) -> Self {
+        self.extra.insert(key, value);
+        self
+    }
```

Comment:
It's common for builder setter methods to just have re-use the field's name (as setting fields is pretty much all a builder does).

Reply:
I think the `add_` prefix is more descriptive. Do you want me to remove it everywhere?

Reply:
I'd personally remove it everywhere, yeah. There are places where it's not already present, so this would make things consistent.

Reply:
Thank you for your feedback, but again, I think the prefix makes the purpose of the method clear and explicit.